packages/material-ui/src/test-utils/createMount.js

```diff
@@ -11,10 +11,16 @@ export default function createMount(options1 = {}) {
   window.document.body.insertBefore(attachTo, window.document.body.firstChild);
 
   const mountWithContext = function mountWithContext(node, options2 = {}) {
+    const { disableUnnmount = false, ...other2 } = options2;
+
+    if (!disableUnnmount) {
+      ReactDOM.unmountComponentAtNode(attachTo);
```

Comment:
Haha, quick and dirty, I was very curious, I was wondering if you would suggest this option. Can it conflict with your test conformance effort?

Reply:
From what I understand, I'm making an implicit behavior explicit. Calling render on an exciting DOM node unmount and mount the new React tree.

Reply:
> Can it conflict with your test conformance effort?

I'm going to merge that PR tomorrow. Then we can rebase here. There's no conflict beyond a diff conflict.

> From what I understand, I'm making an implicit behavior explicit. Calling render on a DOM node unmount and mount the new React tree.

Sure but any behavior from a previous test is now leaking into the next test. This means that test can behave differently depending on the order. It can also cause cascading failures when a test fails

Reply:
I think we can move all the boilerplate into `createMount` and only require a single `const mount = createMount()` per test suite:

```js
export default function createMount(options1 = {}) {
  before(() => {
    // prepare DOM
  })

  afterEach(() => {
    // cleanUp
  })

  after(() => {
    // teardown
  })

  return function mountWithContext(element, options) {
    // enzyme mount
  };
}
```

Reply:
It's less test framework agnostic (although we make the hooks configurable) but the two frameworks I'm used to both implement these hooks (jest and mocha).

Reply:
I will revisit this later. We have some test suites that share state between tests. Unmount between those tests would break the,. Since the solution by @oliviertassinari is working we can use it for now. If notice issues we can prioritize test isolation.

Reply:
It sounds like a great strategy 💯😄👍. I do agree that unmounting after each test is better. I'm just not sure the ROI for our time is here. I share the same pain as you. Our tests isolation is poor. I don't know why. You might have a good explanation for it. From time to time, I have to restart mocha after one test fail, enzyme is going crazy. I'm so used to retard mocha that I don't even pay attention to it anymore. I never had this problem with Jest as it runs each test is a different context.